packages/stream_chat_flutter_core/lib/src/stream_chat_core.dart

```diff
@@ -121,13 +121,14 @@ class StreamChatCoreState extends State<StreamChatCore>
   }
 
   void _subscribeToConnectivityChange([
-    Stream<ConnectivityResult>? connectivityStream,
+    Stream<List<ConnectivityResult>>? connectivityStream,
   ]) {
     if (_connectivitySubscription == null) {
       connectivityStream ??= Connectivity().onConnectivityChanged;
       _connectivitySubscription =
           connectivityStream.distinct().listen((result) {
-        _isConnectionAvailable = result != ConnectivityResult.none;
+        _isConnectionAvailable =
+            result.length == 1 && result.first != ConnectivityResult.none;
```

Comment:
I think if there is more than one element in result it still can be connected. Or even when there is more than one then it's almost certainly connected?

For iOS it can be `something` + `vpn` for example
https://github.com/fluttercommunity/plus_plugins/pull/2599


```suggestion
        _isConnectionAvailable =
            result.length > 1 || (result.length == 1 && result.first != ConnectivityResult.none);
```